lib/services/tokens_service.dart

```diff
@@ -320,23 +213,33 @@ class TokensServiceImpl extends TokensService {
   }
 
   @override
-  Future fetchManualTokens(List<String> indexerIds) async {
+  Future<List<AssetToken>> fetchManualTokens(List<String> indexerIds) async {
     final manuallyAssets = (await _indexer.getNftTokens({"ids": indexerIds}));
 
     //stripe owner for manual asset
     for (var i = 0; i < manuallyAssets.length; i++) {
       manuallyAssets[i].owner = "";
+      manuallyAssets[i].isDebugged = true;
     }
 
     NftCollection.logger.info("[TokensService] "
         "fetched ${manuallyAssets.length} manual tokens. "
         "IDs: $indexerIds");
-    await insertAssetsWithProvenance(manuallyAssets);
+    if (manuallyAssets.isNotEmpty) {
+      await insertAssetsWithProvenance(manuallyAssets);
+    }
+    return manuallyAssets;
   }
 
   @override
-  Future setCustomTokens(List<AssetToken> tokens) async {
-    await _database.assetDao.insertAssets(tokens);
+  Future setCustomTokens(List<AssetToken> assetTokens) async {
+    final tokens = assetTokens.map((e) => Token.fromAssetToken(e)).toList();
+    final assets = assetTokens
+        .where((element) => element.asset != null)
+        .map((e) => e.asset as Asset)
+        .toList();
+    await _database.tokenDao.insertTokens(tokens);
+    await _database.assetDao.insertAssets(assets);
```

Comment:
I think you should check the asset's `lastRefreshTime` before replace the asset. We shouldn't replace the asset if the asset updated time is less than the time saved in DB.